Client/Extensions/UIAlertControllerExtensions.swift

```diff
@@ -118,4 +118,49 @@ extension UIAlertController {
         alert.addAction(noOption)
         return alert
     }
+
+    /**
+     Creates an alert view to warn the user that their logins will either be completely deleted in the 
+     case of local-only logins or deleted across synced devices in synced account logins.
+
+     - parameter deleteCallback: Block to run when delete is tapped.
+     - parameter hasAccount: Boolean indicating the user has a FxA account connected.
```

Comment:
The simplest — and most accurate — thing to do is to ask `SyncablePasswords` if there are any passwords that would propagate as deletions; not only does this avoid trying to extract state from the account, which is complicated and error-prone, but it also correctly muffles the warning if they haven't synced passwords yet, have none saved, or already wiped them. Storage is the source of truth!

For history, #1370 calls this `hasSyncedHistory`, so I'd ape that.

This basically comes down to checking if `loginsM` is non-empty or `loginsL` has anything with a `sync_status` that's not `New`. We're effectively trying to figure out if `SQLiteLogins.removeAll` will leave deletion sentinel rows in `loginsL`.